plugins/solidigm/solidigm-log-page-dir.c

```diff
@@ -0,0 +1,292 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * Copyright (c) 2023 Solidigm.
+ *
+ * Author: karl.dedow@solidigm.com
+ */
+
+#include "solidigm-log-page-dir.h"
+
+#include <errno.h>
+#include <stdio.h>
+
+#include "common.h"
+#include "nvme-print.h"
+
+#include "plugins/ocp/ocp-utils.h"
+
+struct __attribute__((packed)) supported_log_pages {
+	__u32 supported[256];
+};
+
+struct log_description {
+	int lid;
+	const char *description;
+};
+
+static const unsigned char ocp_uuid[NVME_UUID_LEN] = {
+	0xc1, 0x94, 0xd5, 0x5b, 0xe0, 0x94, 0x47, 0x94, 0xa2, 0x1d,
+	0x29, 0x99, 0x8f, 0x56, 0xbe, 0x6f };
+
+static const unsigned char solidigm_uuid[NVME_UUID_LEN] = { 
+	0x96, 0x19, 0x58, 0x6e, 0xc1, 0x1b, 0x43, 0xad, 0xaa, 0xaa,
+	0x65, 0x41, 0x87, 0xf6, 0xbb, 0xb2 
+};
+
+enum solidigm_uuid {
+	NO_UUID,
+	INVALID_UUID,
+	SOLIDIGM_UUID,
+	OCP_UUID,
+};
+
+static int get_uuid_index(struct nvme_id_uuid_list *uuid_list, const unsigned char *uuid)
+{
+	// Some Solidigm drives have swapped UUIDs, so check for that too..
+	unsigned char swapped_uuid[NVME_UUID_LEN] = { 0 };
+	for (int index = NVME_UUID_LEN - 1; index >= 0; index--)
+		swapped_uuid[NVME_UUID_LEN - index - 1] = uuid[index];
+
+	const unsigned char *uuids[2] = { uuid, swapped_uuid };
+
+	for (int index = 0; index < NVME_ID_UUID_LIST_MAX; index++) {
+		for (int count = 0; count < sizeof(uuids) / sizeof(unsigned char *); count++) {
+			if (!memcmp(uuids[count], &uuid_list->entry[index].uuid, NVME_UUID_LEN))
+				return index + 1;
+		}
+	}
+
+	return -1;
+}
+
+static enum solidigm_uuid get_uuid_enum(struct nvme_dev *dev, const int uuid_index)
+{
+	if (uuid_index == 0)
+		return NO_UUID;
+	
+	if (uuid_index < 0 || uuid_index > 127)
+		return INVALID_UUID;
+
+	struct nvme_id_uuid_list uuid_list;
+	int err = nvme_identify_uuid(dev_fd(dev), &uuid_list);
+
+	// If UUID list not supported, then the logs are assumed to be Solidigm (legacy)
+	if (err)
+		return INVALID_UUID;
+
+	int ocp_uuid_index = get_uuid_index(&uuid_list, ocp_uuid);
```

Comment:
use ocp_get_uuid_index()

Reply:
Next commit will remove this logic completely

Reply:
No longer applicable